### ISSUE TITLE
Duplicate claim added in OidcClient.cs.ProcessClaims method when claim valuetype is bool

### ISSUE THREAD

**Issue by roh85** [DuendeArchive/IdentityModel.OidcClient#141]
This issue is directly related to an issue on IdentityServer4.
https://github.com/IdentityServer/IdentityServer4/issues/2982

## Problem
We receive a duplicate email_verified claim in the id_token on the client. But the id_token created by identityserver has this claim only once and is fine. It goes wrong between identityserver and the client in the identitymodel and oidcclient2 component.

I tested this with the latest version of the 9_Combined_AspId_and_EFStorage solution in the IdentityServer4 samples quickstarts as server and a winforms client.

### JObjectExtensions.cs.ToClaims 
The ToClaims method in JObjectExtensions.cs casts all json values to string resulting in lowercase boolean value being cast to 'True' or 'False'

**.ToClaims method, original json**
![image](https://user-images.githubusercontent.com/3336504/61214312-e6c83180-a707-11e9-8a78-a8c5f4cb1683.png)

**.ToClaims method, result**
![image](https://user-images.githubusercontent.com/3336504/61214346-fcd5f200-a707-11e9-8a57-0808f91cce06.png)

### OidcClient.cs.ProcessClaims
The process claim method will combine the userInfoClaims with the userClaims but the ClaimsComparer does not take into account the casing of the claim values. So claims with the same value but different casing are considered different values and the same claim is added twice.
![image](https://user-images.githubusercontent.com/3336504/61214379-0f502b80-a708-11e9-8561-287d9bf768d6.png)

**Comment by leastprivilege** [comment]
I guess this would need a change in the claims comparer logic - which might break other people.

Is this causing a problem?

I opened an issue here

https://github.com/IdentityModel/IdentityModel/issues/232

**Comment by roh85** [comment]
This is not causing an problem right now. When using the `FindFirst` method to filter on claim type it returns only the first match
`var emailVerified = result.User.FindFirst(c => c.Type == JwtClaimTypes.EmailVerified);`
However it would be nice to have this fixed in a future release.

**Comment by brockallen** [comment]
> This is not causing an problem right now. 

How do you then process the value? Do you use `Boolean.Parse`?

**Comment by leastprivilege** [comment]
We found the root cause. Will be fixed in v4 of IdentityModel and then OidcClient at some point as well.
